src/test/scala/com/redis/common/RedisDocker.scala

```diff
@@ -14,58 +20,69 @@ import org.scalatest.time.{Milliseconds, Seconds, Span}
 trait RedisDockerCluster extends RedisContainer {
   that: Suite =>
 
-  protected def redisContainerPort(container: DockerContainer): Int = container.getPorts().futureValue.apply(redisPort)
+  protected def redisContainerPort(container: BaseContainer): Int = container.mappedPort(redisPort)
 
-  protected lazy val runningContainers: List[DockerContainer] = (0 until 4)
-    .map(_ => createContainer())
+  protected def make4Containers: List[Container] = (0 until 4)
+    .map(_ => new Container(createContainer()))
     .toList
 
-  abstract override def dockerContainers: List[DockerContainer] =
-    runningContainers ++ super.dockerContainers
-
+  override val managedContainers: ContainerGroup = ContainerGroup(make4Containers)
 }
 
 trait RedisDocker extends RedisContainer {
   that: Suite =>
 
-  protected lazy val redisContainerPort: Int = runningContainer.getPorts().futureValue.apply(redisPort)
+  implicit lazy val executionContext: ExecutionContext = scala.concurrent.ExecutionContext.global
 
-  private lazy val runningContainer = createContainer()
-
-  abstract override def dockerContainers: List[DockerContainer] =
-    runningContainer :: super.dockerContainers
+  protected def redisContainerPort: Int = managedContainers.container.mappedPort(redisPort)
 
+  override val managedContainers: SingleContainer = SingleContainer(new Container(createContainer()))
 }
 
 trait RedisDockerSSL extends RedisDocker {
   that: Suite =>
 
-  private val certsPath = new File("src/test/resources/certs").getAbsolutePath
+  private lazy val certsPath = new File("src/test/resources/certs").getAbsolutePath
 
-  override protected def baseContainer(name: Option[String]) =
-    DockerContainer("madflojo/redis-tls:latest", name = name)
-      .withVolumes(Seq(VolumeMapping(certsPath, "/certs")))
+  override protected def baseContainer(name: Option[String]): ContainerSpec =
+    name.foldLeft(ContainerSpec("madflojo/redis-tls:latest")
+      .withVolumeBindings(Bind.from(certsPath).to("/certs").build()))(_ withName _)
 }
 
-trait RedisContainer extends DockerKit with DockerTestKit with DockerKitDockerJava with ScalaFutures {
+trait RedisContainer extends DockerTestKitForAll with ScalaFutures {
   that: Suite =>
 
+  def sslContext: Option[SSLContext] = None
+
   implicit val pc: PatienceConfig = PatienceConfig(Span(30, Seconds), Span(100, Milliseconds))
 
   protected val redisContainerHost: String = "localhost"
   protected val redisPort: Int = 6379
 
-  protected def baseContainer(name: Option[String]) = DockerContainer("redis:latest", name=name)
-
+  protected def baseContainer(name: Option[String]): ContainerSpec =
+    name.foldLeft(ContainerSpec("redis:latest"))(_ withName _)
+
+  private def ping: DockerReadyChecker = new DockerReadyChecker {
```

Comment:
Oddly, this is entirely necessary, or the tests are extremely flaky, even with the `LogLineContains` check